### ISSUE TITLE
Type mismatch

### ISSUE THREAD

**Issue by kaminary** [clbr/urlmatch#1]
At [Line 30](https://github.com/clbr/urlmatch/blob/master/simple.c#L30) its not supossed to be `size_t` ?
Correct me if i'm wrong, but I think in 64 bit machines the `size_t` can be bigger than `unsigned` 32 bits.

**Comment by clbr** [comment]
You are right, but who in their right mind uses a string larger than
4gb as the search pattern/target?

Making the variables 64-bit would make it slightly slower for normal
(all) uses. I would say strings larger than 4gb are a special enough
case to warrant custom code anyway.

**Comment by kaminary** [comment]
Yeah, You're correct about the string. Specially when thinking about the URL limit at 2K.
I haven't read all the code, but I think is reasonably to add a check when loading the first time the file where the patterns are, so you can avoid some dumb user from crashing the library.

**Comment by clbr** [comment]
Where do you see a 2k limit?

**Comment by kaminary** [comment]
At RFC [7230](http://tools.ietf.org/html/rfc7230#section-3.1.1).
You can get more info in [this](http://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers) question in stackOverflow
